docs/api_reference/src/quantities.tex

```diff
@@ -146,121 +464,1387 @@
+
+\pnum
+Let \tcode{Proj} be projection used for mapping, and
+let \tcode{T} be the expression template to map from.
+
+\pnum
+\effects
+TBD.
+
+\pnum
+\returns
+TBD.
+\end{itemdescr}
+
+\rSec1[qty.dim]{Dimension}
+
+\rSec2[qty.dim.general]{General}
+
+\pnum
+Subclause \ref{qty.dim} specifies the components
+for defining the dimension of a quantity\irefiev{112-01-11}.
+
+\rSec2[qty.dim.concepts]{Concepts}
+
+\begin{itemdecl}
+template<typename T>
+concept @\deflibconcept{Dimension}@ = @\exposconceptnc{tag-type}@<T> && std::@\stdconcept{derived_from}@<T, @\exposidnc{dimension-interface}@>;
+
+template<typename T>
+concept @\defexposconceptnc{BaseDimension}@ = @\libconcept{Dimension}@<T> && std::@\stdconcept{derived_from}@<T, base_dimension>;
+
+consteval bool @\exposidnc{is-dimension-one}@(std::meta::info type_alias) {
+  return dealias(type_alias) == ^dimension_one;
+}
+
+template<typename T>
+concept @\defexposconceptnc{IsPowerOfDim}@ =
+  (@\exposidnc{is-specialization-of}@(^T, ^power) &&
+   (@\exposconceptnc{BaseDimension}@<typename T::factor> || @\exposidnc{is-dimension-one}@(^typename T::factor)));
+
+template<typename T>
+constexpr bool @\exposidnc{is-per-of-dims}@ = false;
+
+template<typename... Ts>
+constexpr bool @\exposidnc{is-per-of-dims}@<per<Ts...>> =
+  (... && (@\exposconceptnc{BaseDimension}@<Ts> || @\exposidnc{is-dimension-one}@(^Ts) || @\exposconceptnc{IsPowerOfDim}@<Ts>));
+
+template<typename T>
+concept @\defexposconceptnc{DerivedDimensionExpr}@ =
+  @\exposconceptnc{BaseDimension}@<T> || @\exposidnc{is-dimension-one}@(^T) || @\exposconceptnc{IsPowerOfDim}@<T> || @\exposidnc{is-per-of-dims}@<T>;
+
+template<auto D1, auto D2>
+concept @\defexposconceptnc{SameDimension}@ = @\libconcept{Dimension}@<decltype(D1)> && @\libconcept{Dimension}@<decltype(D2)> && D1 == D2;
+
+template<typename T, auto D>
+concept @\deflibconcept{DimensionOf}@ = @\libconcept{Dimension}@<T> && @\libconcept{Dimension}@<decltype(D)> && @\exposconceptnc{SameDimension}@<T{}, D>;
+\end{itemdecl}
+
+\rSec2[qty.dim.types]{Types}
+
+\begin{codeblock}
+namespace mp_units {
+
+struct @\exposidnc{dimension-interface}@ {
+  friend consteval @\libconcept{Dimension}@ auto operator*(@\libconcept{Dimension}@ auto lhs, @\libconcept{Dimension}@ auto rhs)
+  {
+    return @\exposidnc{expr-multiply}@<derived_dimension, struct dimension_one>(lhs, rhs);
+  }
+
+  friend consteval @\libconcept{Dimension}@ auto operator/(@\libconcept{Dimension}@ auto lhs, @\libconcept{Dimension}@ auto rhs)
+  {
+    return @\exposidnc{expr-divide}@<derived_dimension, struct dimension_one>(lhs, rhs);
+  }
+
+  friend consteval bool operator==(@\libconcept{Dimension}@ auto lhs, @\libconcept{Dimension}@ auto rhs)
+  {
+    return ^decltype(lhs) == ^decltype(rhs);
+  }
+};
+
+template<symbol_text Symbol>
+struct @\libglobal{base_dimension}@ : @\exposidnc{dimension-interface}@ {
+  static constexpr auto symbol = Symbol;
```

Comment:
No.
Like _`ratio`_,
it'd be better if the program didn't have access to these interfaces,
as long as they're only needed by the library implementation.
Their use as parameters, initialized from other types, is fine (e.g. `base_dimension<"L">`).

Reply:
I will apply `_XXX_` to those in the repo then. Please keep all members like that expo-only. If I am not wrong, only quantity and quantity_point and traits provide public data members.

Reply:
What about all those member names you've been commenting are exposition-only?
Will you flank them with `_`s?

Reply:
I am not sure if I should do this. I tried a few days ago and added `_` to points origins, but then those stopped to be compatible with `quantity_point` (e.g., `T::quantity_spec`) and I did not want to add more branches or tools to make it work. Maybe I will add `_` is some places but not everywhere because of this.

Any ideas?

Reply:
I think that `quantity` and `quantity_point` should expose all of the properties to the user. We've also agreed that tag types should not do it, and it has sense (e.g., it encourages using non-member functions to obtain some implementation details of tag types), at least until LEWG explicitly asks to do so.